### PR TITLE
Fixed a few fatal errors and warnings in Products CRUD

### PR DESCRIPTION
**Pull request by claudiosanches**

### CODE REVIEW

File: includes/api/class-wc-rest-products-controller.php [hunk 1]
```diff
@@ -1702,8 +1704,7 @@ protected function update_post_meta_fields( $post, $request ) {
 					$this->save_variations_data( $product, $request );
 				} else {
 					// Just sync variations.
-					WC_Product_Variable::sync( $product->get_id() );
-					WC_Product_Variable::sync_stock_status( $product->get_id() );
+					WC_Product_Variable::sync( $product, true );
```
Comment (mikejolley) [inline includes/api/class-wc-rest-products-controller.php]: @justinshreve after your data store change, is this correct? Passing an ID should still work.

Comment (claudiosanches) [inline includes/api/class-wc-rest-products-controller.php]: @mikejolley check this:

```
wp> WC_Product_Variable::sync( 1 )
PHP Fatal error:  Uncaught Error: Cannot pass parameter 1 by reference in phar:///usr/local/bin/wp/php/WP_CLI/REPL.php(37) : eval()'d code:1
```

Need to pass the object now by reference.

Comment (mikejolley) [inline includes/api/class-wc-rest-products-controller.php]: Yeah that cannot happen. @justinshreve Is there a reason this switched to references?

Comment (justinshreve) [inline includes/api/class-wc-rest-products-controller.php]: @mikejolley That wasn't me - it was from your PR the other day #12277. 

https://github.com/woocommerce/woocommerce/pull/12277/files#diff-9d28960c60468fc038da5e6232cf0d8cR709

Comment (claudiosanches) [inline includes/api/class-wc-rest-products-controller.php]: @mikejolley good job troll!

File: includes/class-wc-product-variable.php [hunk 2]
```diff
@@ -503,7 +503,9 @@ protected static function sync_price( &$product ) {
 		if ( ! is_a( $product, 'WC_Product' ) ) {
 			$product = wc_get_product( $product );
 		}
-		$this->data_store->sync_price( $product );
+
+		$data_store = WC_Data_Store::load( 'product_' . $product->get_type() );
```
Comment (justinshreve) [inline includes/class-wc-product-variable.php]: Was `$this->data_store` not already present here from https://github.com/woocommerce/woocommerce/blob/38421fd0953e9987c6daf34dac63655a73c88a0f/includes/abstracts/abstract-wc-product.php#L149 ?

Comment (mikejolley) [inline includes/class-wc-product-variable.php]: Not accessible when called statically.

**Pull request merged**